conn.go

```diff
@@ -376,14 +376,24 @@ func (c *Conn) CloseWithoutFlush() error {
 
 // flush flushes pending data from the buffered writer (c.bw) to the underlying
 // network connection, in case if a buffered writer is present.
-func (c *Conn) flush() {
+func (c *Conn) flush() error {
 	if c.bwPresent {
 		c.bwLock.Lock()
 		defer c.bwLock.Unlock()
 		if c.bw != nil {
-			c.bw.Flush()
+			c.conn.SetWriteDeadline(c.writeDeadline)
+			err := c.bw.Flush()
+			if err != nil {
+				c.writeErrMu.Lock()
+				if c.writeErr != nil {
+					c.writeErr = err
```

Comment:
Do you want the logic the other way around -- `if c.writeErr == nil { ... }`?

Reply:
yup. two of those fixed.